packages/astro/src/content/runtime.ts

```diff
@@ -64,11 +68,19 @@ export function createGetCollection({
 		} else if (collection in dataCollectionToEntryMap) {
 			type = 'data';
 		} else if (store.hasCollection(collection)) {
-			return [...store.values<DataEntry>(collection)].map((entry) => ({
-				...entry,
-				collection,
-				render: () => renderEntry(entry),
-			}));
+			// @ts-expect-error	virtual module
+			const { default: imageAssetMap } = await import('astro:asset-imports');
+
+			return [...store.values<DataEntry>(collection)].map((entry) => {
```

Comment:
nit: maybe we should consider using a `for..of` loop instead of creating a temporary array via spread, and then map it.